pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -608,20 +609,33 @@ func inhaleDeviceConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigCo
 	return parseConfig(config, getconfigCtx, usingSaved)
 }
 
-var lastDevUUIDChange time.Time
+var (
+	lastDevUUIDChange       time.Time
+	potentialUUIDUpdateLock sync.Mutex
+)
 
 // When we think (due to 404) or know that the controller has changed our UUID,
 // ask client to get it so OnboardingStatus can be updated and notified to all agents
 // The controller might do this due to a delete and re-onboard with the same device
 // certificate.
 // We ask client at most every 10 minutes.
-func potentialUUIDUpdate(getconfigCtx *getconfigContext) {
+// We check that another zedclient instance is not running
+func potentialUUIDUpdate(_ *getconfigContext) {
+	potentialUUIDUpdateLock.Lock()
 	if time.Since(lastDevUUIDChange) < 10*time.Minute {
```

Comment:
Why not also check if lastDevUUIDChange is zero/unititialized? 
That should handle the issue at hand (without any lock?)

There might also be a case when we get a 404 (which kicks of zedclient) and 11 minutes later another 404 and for some reason zedclient still is running from the first. So the pidfile check is good insurance.

Reply:
It seems that in case of access to potentialUUIDUpdate from multiple routines we can potentially go through check of time, spend time in pids check and goes into multiple calls to client again. With lock I ensure that we update lastDevUUIDChange from one goroutine and only after that unlock mutex to check from another goroutine.